### PR TITLE
Documentation Improvements Massa-Hash

### PR DESCRIPTION
**Pull request by dr-chain**
- Explaining the need for bitwise XOR
 - Explaining the need for Hash comparison
 - Explaining the motivations for choosing base58 encoding
 - Explaining the motivations for choosing blake3 as the hashing algorithm

* [x] document all added functions

### CODE REVIEW

**Review by Eitu33 — DISMISSED** [review]

**Comment by Eitu33** [comment]
Great 👍

File: massa-hash/src/hash.rs [hunk 1]
```diff
@@ -84,6 +133,16 @@ impl Hash {
     /// let hash = Hash::compute_from(&"hello world".as_bytes());
     /// let serialized: String = hash.to_bs58_check();
     /// ```
+    /// All massa addresses, whenever presented as user-readable texts,
```
Comment (AurelienFT) [inline massa-hash/src/hash.rs:136]: This is not true. Addresses are not only base58 there is prefix also in front of them. I also think that speaking of address in `massa-hash` could lead to confusion as Addresses doesn't exist in this crate.

We could speak only of hashes

File: massa-hash/src/hash.rs [hunk 2]
```diff
@@ -24,6 +51,10 @@ impl PartialOrd for Hash {
     }
 }
 
+/// This function is useful for the BTreeMap where the order of the addresses is to be maintained.
```
Comment (Eitu33) [inline massa-hash/src/hash.rs:54]: Just a tiny specification that it is not only used for addresses

Comment (dr-chain) [reply massa-hash/src/hash.rs]: Fixed

File: massa-hash/src/hash.rs (hunk 1 above)
Comment (dr-chain) [reply massa-hash/src/hash.rs]: Fixed

**Review by AurelienFT — DISMISSED** [review]

File: massa-hash/src/hash.rs [hunk 3]
```diff
@@ -24,6 +51,11 @@ impl PartialOrd for Hash {
     }
 }
 
+/// This function is generally useful for data structures that performs ordering and where hashes are used
+/// as keys. In massa, it is used for the BTreeMap where the order of the addresses is to be maintained.
+/// This function helps to have a single coherent BTreeMap which is then used to perform the draw
+/// See Pos-Worker for more details.
```
Comment (Eitu33) [inline massa-hash/src/hash.rs:57]: What I was trying to say is that we have other use cases in Massa itself than just addresses, like block ids, operation ids, basically anything that has an underlying hash representation.

Comment (dr-chain) [reply massa-hash/src/hash.rs:57]: Ok, I am gonna edit it accordingly.

**Comment by Eitu33** [comment]
@dr-chain Also as a general rule we wait for the comment emitter to resolve the conversation, to make sure that the person is satisfied with the change / response. This avoids creating new conversations or reopening old ones 👍

**Comment by dr-chain** [comment]
bors merge

**Pull request merged**